apps/envited-x-data-space/common/web3/tezos.ts

```diff
@@ -6,17 +6,52 @@ export const initTezos =
   ({ Tezos }: { Tezos: TezosToolkit }) =>
   async () => {
     if (isServer()) {
-      return { Tezos: null, wallet: null }
+      return {
+        Tezos: null,
+        wallet: null,
+        connectWallet: async () => '',
+        disconnectWallet: async () => {},
+        restoreWallet: async () => {},
+      }
     }
 
     const wallet = new (await import('@taquito/beacon-wallet')).BeaconWallet({
       name: process.env.WALLET_NAME || 'Envited Data Space',
       network: { type: 'ghostnet' as any },
       featuredWallets: ['altme', 'temple'],
     })
+
     Tezos.setWalletProvider(wallet)
 
-    return { Tezos, wallet }
+    const connectWallet = async () => {
+      try {
+        await wallet.requestPermissions({ network: { type: 'ghostnet' as any } })
+        const userAddress = await wallet.getPKH()
+        localStorage.setItem('tezos_wallet', userAddress)
```

Comment:
Do we have to deal with localStorage ourselves? Is this not somethin beacon does for us?

Reply:
Correct forgot to take it out